### PR TITLE
[Azure] Improves IPv6 parsing, replaced grok filter with convert 

### LINKED ISSUE #2138: [azure] Handle IPv6 addresses in activity/audit/platform logs
The activity/audit/platform logs data streams use a grok to extract IP addresses but they don't handle IPv6. 

https://github.com/elastic/integrations/blob/917fd079c915b981c699a6e000c5b940c0eccfc7/packages/azure/data_stream/activitylogs/elasticsearch/ingest_pipeline/default.yml#L39-L46

I made changes to the SignInLogs to handle IPv6 so a similar solution could be used for these other data streams. https://github.com/elastic/integrations/pull/1721

> {"type":"illegal_argument_exception","reason":"'2a00' is not an IP string literal."}}, dropping event!
\"ipAddress\":\"2a00:
Although getting other drops such as
{"type":"mapper_parsing_exception","reason":"object mapping for [azure.activitylogs.identity] tried to parse field [identity] as object, but found a concrete value"}, dropping event!

### PR DESCRIPTION
**Pull request by ShourieG**
## What does this PR do?

 1.  Added bug fix for issue #2138 replacing grok filter with convert in 'activitylogs' of azure package
 2. Improved IPv6 handling for other sub-packages : 'auditlogs', 'platformlogs' & 'springcloudlogs' 
 3. Updated ecs.yml file for springcloudlogs to incorporate client.ip as it was missing and is required by definition of the 
     ingest pipeline.
 4. Added edge test case to support IPV6 parsing check  and failures for sub-packages : activitylogs, 'auditlogs', 'platformlogs' & 'springcloudlogs' 

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

 - [x] All tests should pass in azure activitylogs, auditlogs, platformlogs &  springcloudlogs modules , including newly added edge cases IPV6 scenario 

## How to test this PR locally

 Use elastic package to test locally by running command - elastic-package test from inside packages/azure directory

## Related issues

 - Closes #2138

### CODE REVIEW

File: packages/azure/changelog.yml [hunk 1]
```diff
@@ -1,3 +1,8 @@
+- version: "1.1.9"
+  changes:
+    - description: Fix for issue 'azure-issues-2138' by replacing grok filter with convert. Added edge test case to support IPV6 parsing check.
```
Comment (efd6) [inline packages/azure/changelog.yml:3]: Suggest:
```suggestion
    - description: Improve handling of IPv6 IP addresses.
```

File: packages/azure/data_stream/activitylogs/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -33,14 +33,12 @@ processors:
...
-        - "%{IPORHOST:source.ip}:%{INT:source.port:int}"
-        - "%{IPORHOST:source.ip}"
-      ignore_missing: true
-      ignore_failure: true
+  - convert:
+        field: azure.activitylogs.callerIpAddress
+        target_field: source.ip
+        type: ip
+        ignore_missing: true
+        ignore_failure: true
```
Comment (r00tu53r) [inline packages/azure/data_stream/activitylogs/elasticsearch/ingest_pipeline/default.yml:41]: If we're sure it's always an IP (v4 or v6) then there wouldn't be a need for `ignore_failure`. However if there is a possibility that a hostname or some invalid IP address could get in then it might be useful to have `on_failure` and use `set` processor to add that invalid value to `source.address` instead.

File: packages/azure/changelog.yml (hunk 1 above)
Comment (ShourieG) [reply packages/azure/changelog.yml:3]: Sure will update this

File: packages/azure/data_stream/activitylogs/elasticsearch/ingest_pipeline/default.yml (hunk 2 above)
Comment (ShourieG) [reply packages/azure/data_stream/activitylogs/elasticsearch/ingest_pipeline/default.yml:41]: Hey @r00tu53r , if we set an invalid address (that cannot be parsed) as source.address , won't it affect further processing downstream ?

Comment (adriansr) [reply packages/azure/data_stream/activitylogs/elasticsearch/ingest_pipeline/default.yml:41]: source.address is a catch-all field of type keyword, it doesn't have any validation.

https://www.elastic.co/guide/en/ecs/master/ecs-source.html#field-source-address

**Comment by adriansr** [comment]
In order to reference an issue or PR, instead of `azure-issues-2138`, use the `#NNNN` format if the issue belongs to the current repo, or a link if it belongs to a different repo. This way it can be accessed quickly

File: packages/azure/data_stream/activitylogs/elasticsearch/ingest_pipeline/default.yml [hunk 3]
```diff
@@ -33,14 +33,17 @@ processors:
...
-        - "%{IPORHOST:source.ip}"
-      ignore_missing: true
-      ignore_failure: true
+  - convert:
+        field: azure.activitylogs.callerIpAddress
+        target_field: source.ip
+        type: ip
+        ignore_missing: true
+        on_failure:
+        - rename :
```
Comment (r00tu53r) [inline packages/azure/data_stream/activitylogs/elasticsearch/ingest_pipeline/default.yml:42]: extra space:
```suggestion
        - rename: 
```

Comment (ShourieG) [reply packages/azure/data_stream/activitylogs/elasticsearch/ingest_pipeline/default.yml]: removed spaces from all rename processors in latest commit @r00tu53r

**Review by r00tu53r — APPROVED** [review]
LGTM!

**Review by adriansr — APPROVED** [review]
LGTM 🚀

**Review by efd6 — APPROVED** [review]
